src/Core/Services/Implementations/DeviceService.cs

```diff
@@ -16,20 +16,20 @@ public class DeviceService : IDeviceService
         _pushRegistrationService = pushRegistrationService;
     }
 
-    public async Task SaveAsync(Device device)
+    public async Task<Device> SaveAsync(Device device)
     {
+        Device result = null;
         if (device.Id == default(Guid))
         {
-            await _deviceRepository.CreateAsync(device);
+            result = await _deviceRepository.CreateAsync(device);
         }
         else
         {
             device.RevisionDate = DateTime.UtcNow;
             await _deviceRepository.ReplaceAsync(device);
+            result = device;
         }
-
-        await _pushRegistrationService.CreateOrUpdateRegistrationAsync(device.PushToken, device.Id.ToString(),
-            device.UserId.ToString(), device.Identifier, device.Type);
+        return result;
     }
 
     public async Task ClearTokenAsync(Device device)
```

Comment:
It looks like with the removal of the `/clear-token`, this isn't used anymore. Is that correct? If so, should we remove this method and definition in IDeviceService?

Reply:
Yes, thank you!  I just updated with another commit.